### PR TITLE
Added slow but thorough matrix benchmarks

### PR DESCRIPTION
**Pull request by Pristine-Cat**
These were requested to test the new matrix intermediate simplification in sympy PR #18572.

### CODE REVIEW

File: slow_benchmarks/matrices.py [hunk 1]
```diff
@@ -0,0 +1,455 @@
+from sympy import Symbol, Matrix, symbols, SparseMatrix, Abs
+from sympy import I, S, simplify
+from sympy.simplify.simplify import simplify
+
+_TEST_SIMPLIFY = False # test simplify after operation?
```
Comment (oscarbenjamin) [inline slow_benchmarks/matrices.py]: Why is this needed?

Comment (Pristine-Cat) [reply slow_benchmarks/matrices.py]: In the original benchmark since dotprodsimp was optional this was for comparing `func(dotprodsimp=True)` vs. `simplify(func())`. I left the code in just in case at some point someone wants to make the same comparison.

File: slow_benchmarks/matrices.py [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+DDS = SparseMatrix(DD)
+
+O4 = Matrix(4, 1, [1, 1, 1, 1])
+
+
+class TimePow4:
+    def time_A(self): A**4
+    def time_B(self): B**4
+    def time_C(self): C**4
+    def time_D(self): D**4
```
Comment (oscarbenjamin) [inline slow_benchmarks/matrices.py]: Maybe could do
```
class _TimeMatrixBase:
    def time_A(self): self.func(A)
    def time_B(self): self.func(B)
    def time_C(self): self.func(C)
    def time_D(self): self.func(D)

class TimePow4(_TimeMatrixBase):
    def func(self, M):
        return M**4
```

Comment (Pristine-Cat) [reply slow_benchmarks/matrices.py]: Probably, but at this point with my potentially limited time I am focusing on the actual matrix code...

**Comment by moorepants** [comment]
Note that we specifically want to avoid slow benchmarks. I suggest that benchmarks should be under a second here: https://github.com/sympy/sympy_benchmarks/issues/8

**Comment by Pristine-Cat** [comment]
Are the benchmarks in the `slow_benchmarks` directory run by default? I thought I had read than no, so that is where these reside.

**Comment by moorepants** [comment]
I see. I guess those are not run, see https://github.com/sympy/sympy_benchmarks/issues/35.
